modules/mattermost-notification-workflowoperation/src/main/java/org/opencastproject/workflow/handler/mattermost/notification/MattermostNotificationWorkflowOperationHandler.java

```diff
@@ -0,0 +1,298 @@
+      }
+
+      request.setEntity(new UrlEncodedFormEntity(params, "UTF-8"));
+    } catch (UnsupportedEncodingException e) {
+      throw new WorkflowOperationException(
+              "Error happened during the encoding of the event parameter as form parameter: {}", e);
+    }
+
+    // Execute the request
+    if (!executeRequest(request, maxRetry, timeout, INITIAL_SLEEP_TIME)) {
+      throw new WorkflowOperationException(format("Notification could not be delivered to %s", urlPath));
+    }
+
+    return createResult(workflowInstance.getMediaPackage(), Action.CONTINUE);
+  }
+
+  /**
+   * Gets a notification message with placeholders and substitute them with corresponding meta-data of workflowInstance.
+   * The resulting String is transformed to a Json-String
+   *
+   * @param s                The notification message to transform to Json-String
+   * @param workflowInstance The workflowInstance which getting metadata from
+   * @return JSON-String containing the information of the workflowInstance
+   */
+  private String makeJson(String s, WorkflowInstance workflowInstance) {
+    s = s.replace("%t", checkIfNull(workflowInstance.getTitle(), "Title"));
+    s = s.replace("%i", String.valueOf(workflowInstance.getId()));
+    s = s.replace("%s", String.valueOf(workflowInstance.getState()));
+    s = s.replace("%o", String.valueOf(workflowInstance.getCurrentOperation().getId()));
+    s = s.replace("%I", checkIfNull(workflowInstance.getMediaPackage().getIdentifier(), "Mediapackage-ID"));
+    s = s.replace("%T", checkIfNull(workflowInstance.getMediaPackage().getTitle(), "Mediapackage-Title"));
+    s = s.replace("%c", checkIfNull(workflowInstance.getMediaPackage().getContributors(), "Contributors"));
+    s = s.replace("%C", checkIfNull(workflowInstance.getMediaPackage().getCreators(), "Creators"));
+    s = s.replace("%D", checkIfNull(workflowInstance.getMediaPackage().getDate(), "Date"));
+    s = s.replace("%d", checkIfNull(workflowInstance.getMediaPackage().getDuration(), "Duration"));
+    s = s.replace("%l", checkIfNull(workflowInstance.getMediaPackage().getLanguage(), "Language"));
+    s = s.replace("%L", checkIfNull(workflowInstance.getMediaPackage().getLicense(), "License"));
+    s = s.replace("%S", checkIfNull(workflowInstance.getMediaPackage().getSeriesTitle(), "Series-Title"));
+
+    JsonObject json = new JsonObject();
+    json.addProperty("text", s);
+    GsonBuilder builder = new GsonBuilder();
+    Gson gson = builder.create();
+    return gson.toJson(s);
+  }
+
+  /**
+   * Checks if an object is null. If an object is null, then method returns not defined, else it returns object as a
+   * String
+   *
+   * @param o The object to check
+   * @param s The name of metadata to check
+   * @return String containing the transformed object
+   */
+  private String checkIfNull(Object o, String s) {
+
+    if(o == null) {
```

Comment:
That results in a build error:

[ERROR] /home/travis/build/opencast/opencast/modules/mattermost-notification-workflowoperation/src/main/java/org/opencastproject/workflow/handler/mattermost/notification/MattermostNotificationWorkflowOperationHandler.java:239:7: 'if' is not followed by whitespace. [WhitespaceAround]
Audit done.